seacrowd/sea_datasets/alorese/alorese.py

```diff
@@ -0,0 +1,269 @@
+import pandas as pd
+
+from seacrowd.sea_datasets.alorese.alorese_url import _URLS_DICT
+from seacrowd.utils import schemas
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Licenses, Tasks
+
+_CITATION = """\
+@article{Moro2018-ms,
+  title     = "The plural word hire in alorese: Contact-induced change from
+               neighboring Alor-pantar languages",
+  author    = "Moro, Francesca R",
+  journal   = "Oceanic Linguistics",
+  publisher = "University of Hawai'i Press",
+  volume    =  57,
+  number    =  1,
+  pages     = "177--198",
+  year      =  2018,
+  language  = "en"
+}
+"""
+
+_DATASETNAME = "alorese"
+_DESCRIPTION = """\
+ Alorese Corpus is a collection of language data in a couple of Alorese variation (Alor and Pantar Alorese). The collection is available in video, audio, and text formats with genres ranging from Experiment or task, Stimuli, Discourse, and Written materials.
+"""
+_HOMEPAGE = "	https://hdl.handle.net/1839/e10d7de5-0a6d-4926-967b-0a8cc6d21fb1"
+_LANGUAGES = ["aol", "ind"]
+_LICENSE = Licenses.UNKNOWN.value
+_LOCAL = False
+
+_URLS = _URLS_DICT
+
+_SUPPORTED_TASKS = [Tasks.SPEECH_RECOGNITION, Tasks.MACHINE_TRANSLATION]
+
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+
+class AloreseDataset(datasets.GeneratorBasedBuilder):
+    """Alorese Corpus is a collection of language data in a couple of Alorese variation (Alor and Pantar Alorese). The collection is available in video, audio, and text formats with genres ranging from Experiment or task, Stimuli, Discourse, and Written materials."""
+
+    SUBSETS = ["t2t", "sptext", "sptext_trans"]
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_{subset}_source",
+            version=datasets.Version(_SOURCE_VERSION),
+            description=f"{_DATASETNAME} source schema for {subset} subset",
+            schema="source",
+            subset_id=f"{_DATASETNAME}_{subset}",
+        )
+        for subset in SUBSETS
+    ] + [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_t2t_seacrowd_t2t",
+            version=datasets.Version(_SEACROWD_VERSION),
+            description=f"{_DATASETNAME} SEACrowd schema for t2t subset",
+            schema=f"seacrowd_t2t",
+            subset_id=f"{_DATASETNAME}_t2t",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_sptext_seacrowd_sptext",
+            version=datasets.Version(_SEACROWD_VERSION),
+            description=f"{_DATASETNAME} SEACrowd schema for sptext subset",
+            schema=f"seacrowd_sptext",
+            subset_id=f"{_DATASETNAME}_sptext",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_sptext_trans_seacrowd_sptext",
+            version=datasets.Version(_SEACROWD_VERSION),
+            description=f"{_DATASETNAME} SEACrowd schema for sptext_trans subset",
+            schema=f"seacrowd_sptext",
+            subset_id=f"{_DATASETNAME}_sptext_trans",
+        ),
+    ]
```

Comment:
may I ask few qns here?

1. overall, the schema we will use in here are T2T for translation of transcription of same audio from Indonesian to Alorese, and the SPText is the ASR version, right?

2. Which language does the audio content has? Indonesian or Alorese? bcs when I looked the code, there's no clear way to tell which lang is available in the transcription (and it's crucial to correctly map the Audio to the correct Transcriptions)

3. For the config name below:
`{_DATASETNAME} _sptext_seacrowd_sptext` and `{_DATASETNAME}_t2t_seacrowd_t2t`, may I know why the naming isn't something like `{_DATASETNAME}_seacrowd_sptext` and `{_DATASETNAME}_seacrowd_t2t`? any justifications here?

4. I saw in the source, the info on `speaker_id` went missing? I thought `source` config should include all columns and informations (and stitch them appropriately if the data scattered into different files and configs, like what you did in your code)

Reply:
Hi @sabilmakbar , thanks for the review and questions!

1. Yes
2. The audio is in alorese, the lang available in transcription is denoted by either `sptext` for alorese or `sptext_trans` for indonesian. The mapping is done in this part
<img width="1095" alt="image" src="https://github.com/SEACrowd/seacrowd-datahub/assets/65403659/0236664f-9335-4124-a3af-4de56c3e86ab">

3. Because of the subset naming, the testing code (and dataset naming format) that was constructed needs to be in `<DATASET_NAME>_<SUBSET_NAME>_seacrowd_`<SCHEMA>`. It's just the subset name happens to be same as the schema, so it might be confusing. Happy to change if it might be needed.

4. Thanks for the input! Please review the latest commit as I have done the nitpick.

Reply:
thanks for the prompt reply, @patrickamadeus!

> The audio is in alorese, the lang available in transcription is denoted by either sptext for alorese or sptext_trans for indonesian. The mapping is done in this part

Idk whether we need to create a schema of alorese audio against Indonesian text for the SPText schema. my personal opinion is to remove it since it could be very misleading (ASR schema should provide a text which is an actual transcription as is, not the translated ones)


> Because of the subset naming, the testing code (and dataset naming format) that was constructed needs to be in <DATASET_NAME>_<SUBSET_NAME>_seacrowd_`. It's just the subset name happens to be same as the schema, so it might be confusing. Happy to change if it might be needed.

If I understand this correctly, the subsets for this dataset are only the Alorese version and Indonesian version. The SPText and T2T don't fit properly to the definition of "subset" of a dataset, as only the schema is different.


wdyt? @ljvmiranda921 @holylovenia @SamuelCahyawijaya

Reply:
Ah I see, hmm, does it mean we should have a separate data loaders for the SPText and the T2T versions? I don't have a particularly strong opinion on any approach.

Reply:
> Ah I see, hmm, does it mean we should have a separate data loaders for the SPText and the T2T versions?

No, we can have it in a single dataloader; I think the subset and configuration naming should be modified slightly.

> The audio is in alorese, the lang available in transcription is denoted by either sptext for alorese or sptext_trans for indonesian.

The text information provided in this dataset is sequential; i.e., for every audio file, there is a sequence of annotated texts with their start & end timestamps.

For the Alorese language of text and audio, this can be put in ASR schema (or even a sequential audio split to its annotated text if we want to refine it further).

However, I don't think we should create an ASR schema for the Alorese audio and translated Indonesian annotation since the audio and text are in different languages.

And for T2T of Alorese and Indonesian translation, the existing implementation is correct, just need to reconstruct the configs list.

Therefore, my proposed configs are:
1. Source -- containing Alorese Audio, Alorese Annotation (and its timestamp), and Indonesian Annotation (and its timestamp too)
2. SPText -- containing Alorese Audio & its Annotation (the annotation could be combined into single text per audio -- as previously implemented or recreated using new sequenced schema of text)
3. T2T -- containing Alorese Annotation & translated Indonesian Annotation (we can leave this as full-text translation, not word-to-word or phrase-to-phrase translation)

Reply:
Hi! Thanks for the suggestions @sabilmakbar !

> Btw, just confirming, do the audio recordings and the transcriptions match word-for-word?

Yes! There are multiple timestamps to indicate when each word is spoken. @holylovenia , to be honest I haven't reviewed substantial sample from the data to determine whether it matched word-for-word or no, but as I listened to the first 10 seconds of 1 particular example, it perfectly matched.

If there is no further suggestion or comment, I will implement the change maximum this weekend, got a bunch of stuff to do first.

Reply:
Hi @sabilmakbar ! Could you please check on the latest commit? I have done the revision 👍.

> SPText -- containing Alorese Audio & its Annotation (the annotation could be combined into single text per audio -- as previously implemented or recreated using new sequenced schema of text)

For this one, I went on using the previous implementation first.